test/e2e/suite/internal/client/client.go

```diff
@@ -70,35 +73,22 @@ func NewCertManagerClient(endpoint string, tls bool, rootCert []byte, clusterID
 }
 
 // CSR Sign calls cert-manager istio-csr to sign a CSR.
-func (c *certmanagerClient) CSRSign(ctx context.Context, reqID string, csrPEM []byte, token string,
-	certValidTTLInSec int64) ([]string /*PEM-encoded certificate chain*/, error) {
+func (c *certmanagerClient) CSRSign(csrPEM []byte, certValidTTLInSec int64) ([]string, error) {
 	req := &securityapi.IstioCertificateRequest{
 		Csr:              string(csrPEM),
 		ValidityDuration: certValidTTLInSec,
 	}
-
-	if token != "" {
-		// add Bearer prefix, which is required by cert-manager istio agent.
-		token = bearerTokenPrefix + token
-		ctx = metadata.NewOutgoingContext(ctx, metadata.Pairs("Authorization", token, "ClusterID", c.clusterID))
-	} else {
-		// This may use the per call credentials, if enabled.
```

Comment:
Question: is this an option that was never implemented in istio-csr anyway? (I see all the e2e tests always used a token).

Reply:
Correct. This file was adapted from [istio's own caClient](https://github.com/istio/istio/blob/c3d92d81db5dd5b2cba7211db25f0ab8b51a6d3b/security/pkg/nodeagent/caclient/providers/citadel/client.go#L41). Removing unused bits is always good :factory_worker: